### PR TITLE
DOP-1177: Catch Parser Failures

### PR DESCRIPTION
**Pull request by madelinezec**
Followed the same routine as in [DOP-64 ](https://github.com/mongodb/docs-worker-pool/pull/198)

In brief: 

I generate a parser failure by changing the `name` field in the docs-repo snooty.toml to `invalid`. Commit and push this change to enqueue a job to watch the server report the failure. 

I tested next-gen-publish by running up our staging ec2 instance and doing `make next-gen-publish` on docs-node. The failure was generated. 

TODO: 
inspect why docs and docs-tutorials do not have proper failure reporting. Their messages do not include the failures and stack straces, but rather this: 
```
Your Job (Github Push: madelinezec/docs-tutorials) has successfully been added to the queue.
Your Job (Github Push: madelinezec/docs-tutorials) is now being processed.
Your Job (Github Push: madelinezec/docs-tutorials) is now being processed.
Your Job (Github Push: madelinezec/docs-tutorials) is now being processed.
Your Job (Github Push: madelinezec/docs-tutorials) is now being processed.
Your Job (Github Push: madelinezec/docs-tutorials) has failed 3 times and will not be placed back in the queue.
```
This issue has been captured in this ticket: https://jira.mongodb.org/browse/DOP-1206

### CODE REVIEW

File: makefiles/Makefile.docs-mongodb-shell [hunk 1]
```diff
@@ -65,7 +65,12 @@ deploy-search-index:
 	
 next-gen-html:
 	# snooty parse and then build-front-end
-	@-echo ${SNOOTY_DB_PWD} | snooty build "${REPO_DIR}" "mongodb+srv://${SNOOTY_DB_USR}:@cluster0-ylwlz.mongodb.net/snooty?retryWrites=true" || exit 0;
+	@echo ${SNOOTY_DB_PWD} | snooty build "${REPO_DIR}" "mongodb+srv://${SNOOTY_DB_USR}:@cluster0-ylwlz.mongodb.net/snooty?retryWrites=true";
+	if [ $? -eq 1 ]; then \
```
Comment (i80and) [inline makefiles/Makefile.docs-mongodb-shell:69]: Does this need to be `$$?`?

File: makefiles/Makefile.docs-mongodb-vscode [hunk 2]
```diff
@@ -56,7 +56,12 @@ deploy: build/public
 
 next-gen-html:
 	# snooty parse and then build-front-end
-	@-echo ${SNOOTY_DB_PWD} | snooty build "${REPO_DIR}" "mongodb+srv://${SNOOTY_DB_USR}:@cluster0-ylwlz.mongodb.net/snooty?retryWrites=true" || exit 0;
+	@echo ${SNOOTY_DB_PWD} | snooty build "${REPO_DIR}" "mongodb+srv://${SNOOTY_DB_USR}:@cluster0-ylwlz.mongodb.net/snooty?retryWrites=true";
+	if [ $? -eq 1 ]; then \
```
Comment (i80and) [inline makefiles/Makefile.docs-mongodb-vscode:60]: ditto

File: makefiles/Makefile.docs-mongodb-vscode [hunk 3]
```diff
@@ -71,7 +76,12 @@ next-gen-stage: ## Host online for review
 
 next-gen-publish:
 	# snooty parse and then build-front-end
-	@-echo ${SNOOTY_DB_PWD} | snooty build "${REPO_DIR}" "mongodb+srv://${SNOOTY_DB_USR}:@cluster0-ylwlz.mongodb.net/snooty?retryWrites=true" || exit 0;
+	@echo ${SNOOTY_DB_PWD} | snooty build "${REPO_DIR}" "mongodb+srv://${SNOOTY_DB_USR}:@cluster0-ylwlz.mongodb.net/snooty?retryWrites=true"; \
+	if [ $? -eq 1 ]; then \
```
Comment (i80and) [inline makefiles/Makefile.docs-mongodb-vscode:80]: ditto

File: makefiles/Makefile.docs-k8s-operator [hunk 4]
```diff
@@ -78,7 +78,13 @@ publish:
...
 next-gen-html:
 	# snooty parse and then build-front-end
-	@-echo ${SNOOTY_DB_PWD} | snooty build ${REPO_DIR} "mongodb+srv://${SNOOTY_DB_USR}:@cluster0-ylwlz.mongodb.net/snooty?retryWrites=true" || exit 0;
+	@echo ${SNOOTY_DB_PWD} | snooty build ${REPO_DIR} "mongodb+srv://${SNOOTY_DB_USR}:@cluster0-ylwlz.mongodb.net/snooty?retryWrites=true";
+	if [ $? -eq 1 ]; then \
+		exit 1; \
+	else \
+		exit 0; \
+	fi
+	#rsync -az --exclude '.git' ${REPO_DIR}/../../snooty ${REPO_DIR};
```
Comment (skerschb) [inline makefiles/Makefile.docs-k8s-operator:87]: I notice that some of these Makefiles have rsync for the copy and some of them use the cp. Some of them appear to have both. Can we standardize?

**Pull request merged**